spec/unit/device_spec.lua

```diff
@@ -166,17 +166,82 @@ describe("device module", function()
     end)
 
     describe("kindle", function()
-        it("should initialize voyage without error", function()
-            io.open = function(filename, mode)
+        local function make_io_open_kindle_model_override(model_no)
+            return function(filename, mode)
                 if filename == "/proc/usid" then
                     return {
-                        read = function() return "B013XX" end,
+                        read = function() return model_no end,
                         close = function() end
                     }
                 else
                     return iopen(filename, mode)
                 end
             end
+        end
+
+        insulate("without framework", function()
+            local mock_lipc = {
+                init = function()
+                    return {
+                        set_int_property = mock(function() end),
+                        get_int_property = function() return 0 end,
+                        get_string_property = function() return "string prop" end,
+                        set_string_property = function() end,
+                        register_int_property = function() return {} end,
+                        close = function () end,
+                    }
+                end
+            }
+            package.loaded['liblipclua'] = mock_lipc
```

Comment:
nit

Reply:
(Meaning double quotes instead of single quotes ;)).